controllers/teleport_controller/teleport_controller.cpp

```diff
@@ -0,0 +1,138 @@
+// File:          teleport_controller.cpp
+// Date:          23/03/2021
+// Description:   Method to teleport robot around the playing field for the purposes of
+//                    gathering training data
+// Author:        Lachlan Court
+// Modifications:
+
+
+#include <RobotisOp2GaitManager.hpp>
+#include <RobotisOp2MotionManager.hpp>
+#include <webots/Supervisor.hpp>
+#include <iostream>
+#include <math.h>
+#include <time.h> 
+
+// All the webots classes are defined in the "webots" namespace
+using namespace webots;
+
+/*
+// Rotations matrix for 90 degree rotations
+static const int rotationsSize = 4;
+static const double rotations[rotationsSize][4] = {{1, 0, 0, 1.57081},
+                                                  {0.577353, 0.577345, 0.577353, 2.0944},
+                                                  {1.69526e-09, 0.707103, 0.707111, 3.14159},
+                                                  {-0.577352, 0.577347, 0.577352, -2.09441}};
+
+*/
+
+// Rotations matrix for 45 degree rotations
+static const int rotationsSize = 8;                                     
+static const double rotations[rotationsSize][4] = {{1, 0, 0, 1.57081},
+                                                  {0.867737, 0.343283, 0.359429, 1.75753},
+                                                  {0.577353, 0.577345, 0.577353, 2.0944},
+                                                  {0.286947, 0.661632, 0.692752, 2.60662},
+                                                  {1.69526e-09, 0.707103, 0.707111, 3.14159},
+                                                  {-0.286948, 0.661631, 0.692752, -2.60661},
+                                                  {-0.577352, 0.577347, 0.577352, -2.09441},
+                                                  {0.867739, -0.343277, -0.359429, 1.75753}};                               
+
+
+/* This is the main program of the supervisor. It creates an instance of the Supervisor
+ *     instance, launches its function(s) and destroys it at the end of the execution.
+ */
+int main() {
+    std::cout << "a" << std::endl;
+    // create the Supervisor instance and assign it to a robot
+    Supervisor* supervisor = new Supervisor();
+    Node* target           = supervisor->getFromDef("RED_1");
+    std::cout << "b" << std::endl;
+    // Get the time step of the current world.
+    int timeStep = (int) supervisor->getBasicTimeStep();
+
+    std::cout << "c" << std::endl;
+
+    // Add the motion manager to make the robot stand up and look around
+    managers::RobotisOp2MotionManager* mMotionManager = new managers::RobotisOp2MotionManager(supervisor);
+    mMotionManager->playPage(54);
+
+    // Generate random seed
+    srand(time(NULL));
```

Comment:
If we are generating a random seed, is it necessary to set one at all? (I am not sure)

Reply:
Also, use the c++ random number library.

Reply:
Depends, if we only ever use the default seed it would run the same every time we started it up wouldn't it? In the interests of getting a broad data set I thought that would be best
I didn't realise there was a random number library, I'll look into it

Reply:
well it depends what the default behaviour is, as in, what happens if we don't give it a seed at all? most programs that can be seeded use a random seed if you don't give them a seed explicitly. It's fine to seed it anyway of course.

Reply:
I will run a few tests and see if it makes a difference, if not I'll remove it

Reply:
> For every different seed value used in a call to srand, the pseudo-random number generator can be expected to generate a different succession of results in the subsequent calls to rand.
> 
> Two different initializations with the same seed will generate the same succession of results in subsequent calls to rand.

In other words, setting the seed to a specific value will effectively guarantee the same random sequence of numbers. Setting the seed to a different value (like the current time) will ensure you get a different sequence of random numbers every time you run the program (i.e. every time you run you should expect different sequences of random behaviour)

Reply:
doing some reading... why the f does C++ rand automatically use `seed == 1` without an `srand()` call wtf ahaha. Yea, you'll need to seed it for sure Lachlan

Reply:
Well, if you're using c++11 or higher, you shouldn't ever be using std::rand anyway. It has problems.

Modern method is to use the new c++ random functionality to do this.

Pick a distribution and then do what it does below, make a pseudorandom number generator based off a true random source.
https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution

Reply:
My only experience with c++ is 98 thanks to SENG1120 😆  I haven't delved into the wonders that are c++<literally anything else> so I'm excited to learn this better way of doing random numbers!